### PR TITLE
added timeout and connection timeout property to guzzle client

### PR DESCRIPTION
**Pull request by rameshdahiya**
https://urbansportsclub.atlassian.net/browse/TECH-2710

### CODE REVIEW

File: src/ApiClient.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 namespace OneFit\GraphQLClient;
 
 use GuzzleHttp\ClientInterface;
+use Illuminate\Support\Arr;
```
Comment (ricardofiorani) [inline src/ApiClient.php:6]: If you move this to the Service provider then this class should be easier to test and have less static dependencies.

File: src/ApiClient.php [hunk 2]
```diff
@@ -20,12 +21,14 @@ class ApiClient extends GraphQLClient
...
      * @param string $baseUrl
-     * @param string $token
+     * @param array $requestOptions
      */
-    public function __construct(ClientInterface $httpClient, string $baseUrl)
+    public function __construct(ClientInterface $httpClient, string $baseUrl, array $requestOptions = [])
     {
         $this->httpClient = $httpClient;
         $this->baseUrl = $baseUrl;
+        $this->connectionTimeout = Arr::get($requestOptions,'connect_timeout', 0);
```
Comment (ricardofiorani) [inline src/ApiClient.php:30]: why not just `$requestOptions['connection_timeout']` ?

**Review by ricardofiorani — COMMENTED** [review]
Hi Ramesh, please take my CR with a grain of salt as I'm not completely familiar with the whole stack. 
I added some comments without requesting changes just get discussion about it and to get familiar.

File: src/GraphQLClient.php [hunk 3]
```diff
@@ -25,6 +29,16 @@ abstract class GraphQLClient
      */
     protected $options = [];
 
+    /**
+     * total timeout of the request in seconds
```
Comment (osalimi) [inline src/GraphQLClient.php:33]: types ?

File: src/ApiClient.php (hunk 1 above)
Comment (rameshdahiya) [reply src/ApiClient.php:6]: ~~Arr class is part of Illuminate package of laravel it's an helper class, and suppose to be use this way. As the helper class injected using the composer https://github.com/laravel/framework/blob/5.8/composer.json#L97
So we don't need to inject it separately~~

File: src/ApiClient.php (hunk 2 above)
Comment (rameshdahiya) [reply src/ApiClient.php:30]: Could be an option too, I am using helper method to have clean code.

File: src/GraphQLClient.php (hunk 3 above)
Comment (rameshdahiya) [inline src/GraphQLClient.php:33]: its float , I will mention this in doc block

File: src/ApiClient.php (hunk 1 above)
Comment (rameshdahiya) [reply src/ApiClient.php:6]: My bad @ricardofiorani the above explanation was the wrong explanation to you comment. I Agree with you, static dependency we should avoid, only if those you wanted to mock. However i don't want to mock Arr at all, its part of Illuminate/Support ,where it already have its test cases handled. And I wanted to use as it is, even for my test.

**Review by ricardofiorani — APPROVED** [review]

**Review by osalimi — APPROVED** [review]

**Pull request merged**